Floofbot/Modules/Tag.cs

```diff
@@ -7,191 +7,188 @@
-                    }
+        [Command("add")]
+        [Priority(0)]
+        [RequireUserPermission(GuildPermission.AttachFiles)]
+        public async Task Add(string Tag, [Remainder] string Content = null)
+        {
+            try {
+                if (Content != null) {
+                    Regex rgx = new Regex("[^a-zA-Z0-9 -]");
+                    Tag = rgx.Replace(Tag, "").ToLower();
+                    string sql = @"INSERT into Tags (TagID,UserID, Content)
+                            VALUES ($TagId, $UserID, $Content)";
+
+                    string tagId = $"{Tag.ToString()}:{Context.Guild.Id.ToString()}";
+                    SqliteCommand command = new SqliteCommand(sql, dbConnection);
+                    command.Parameters.Add(new SqliteParameter("$TagId", tagId));
+                    command.Parameters.Add(new SqliteParameter("$UserID", Context.User.Id.ToString()));
+                    command.Parameters.Add(new SqliteParameter("$Content", Content));
+                    var result = command.ExecuteScalar();
+
+                    await SendEmbed(CreateDescriptionEmbed($"💾 Added Tag `{Tag}`"));
                 }
-                catch (SqliteException) {
-                    await sendEmbed(createDescriptionEmbed($"💾 Tag `{Tag}` Already Exists"));
+                else {
+                    await SendEmbed(CreateDescriptionEmbed($"💾 Usage: `tag add [name] [content]`"));
                 }
             }
-
-            [Command("add")]
-            [Priority(1)]
-            public async Task Add()
-            {
-                await sendEmbed(createDescriptionEmbed($"💾 Usage: `tag add [name] [content]`"));
+            catch (SqliteException) {
+                await SendEmbed(CreateDescriptionEmbed($"💾 Tag `{Tag}` Already Exists"));
             }
+        }
 
-            //TODO: Fix me
-            [Command("list")]
-            public async Task ListTags([Remainder] string Content = null)
-            {
-                string guildId = $"{Context.Guild.Id}";
-                string sql = $"SELECT tagID FROM Tags WHERE tagID LIKE '%:$guildId%'";
-                SqliteCommand command = new SqliteCommand(sql, dbConnection);
-                command.Parameters.Add(new SqliteParameter("$guildId", guildId));
-                var result = command.ExecuteReader();
+        [Command("add")]
+        [Priority(1)]
+        public async Task Add()
+        {
+            await SendEmbed(CreateDescriptionEmbed($"💾 Usage: `tag add [name] [content]`"));
+        }
 
-                List<string> tags = new List<string>();
-                List<string> pages = new List<string>();
+        //TODO: Fix me
+        [Command("list")]
+        public async Task ListTags([Remainder] string Content = null)
+        {
+            string guildId = $"{Context.Guild.Id}";
+            string sql = $"SELECT tagID FROM Tags WHERE tagID LIKE '%:$guildId%'";
```

Comment:
I'm guessing you are putting the guildID as part of the tagID

I am not sure I like this approach, I think it would be better to have a column for server ID

Reply:
I'll handle this in a future PR as part of #14 .